File: php/src/Creational/FactoryMethod/HonoluluSoupFactoryMethodSubclass.php
<?

#//HonoluluSoupFactoryMethodSubclass.java - Two of Two Subclass Factory Methods

require_once 'SoupFactoryMethod.php';

class HonoluluSoupFactoryMethodSubclass extends SoupFactoryMethod
{

	function makeBuffetName() { return "Honolulu Soup Buffet"; }
	function makeClamChowder() { return new HonoluluClamChowder(); } 
	function makeFishChowder() { return new HonoluluFishChowder(); }

}

File: go/creational/FactoryMethod/HonoluluSoupFactoryMethodSubclass.go
//HonoluluSoupFactoryMethodSubclass - Two of Two Subclass Factory Methods


package factorymethod



public class HonoluluSoupFactoryMethodSubclass extends SoupFactoryMethod {
    public string makeBuffetName() {
        return "Honolulu Soup Buffet"
    }
    public ClamChowder makeClamChowder() {
        return new HonoluluClamChowder()
    }
    public FishChowder makeFishChowder() {
        return new HonoluluFishChowder()
    }
}
